tests/shared/src/test/scala/zio/schema/StandardTypeCoerceSpec.scala

```diff
@@ -0,0 +1,169 @@
+
+          assertTrue(
+            coercion.get("true") == Right(true),
+            coercion.get("false") == Right(false),
+            coercion.get("hello") == Left("String hello could not be coerced into a Boolean"),
+            coercion.get("1") == Left("String 1 could not be coerced into a Boolean")
+          )
+        },
+        test("-> LocalDate") {
+          val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
+          val coercion  = StandardType.StringType.coerce(StandardType.LocalDateType(formatter))
+
+          assertTrue(
+            coercion.get("2020-12-31") == Right(java.time.LocalDate.parse("2020-12-31", formatter)),
+            coercion.get("hello") == Left("String hello could not be coerced into a LocalDate")
+          )
+        },
+        test("-> LocalTime") {
+          val formatter = DateTimeFormatter.ofPattern("HH:mm:ss")
+          val coercion  = StandardType.StringType.coerce(StandardType.LocalTimeType(formatter))
+
+          assertTrue(
+            coercion.get("12:34:56") == Right(java.time.LocalTime.parse("12:34:56", formatter)),
+            coercion.get("hello") == Left("String hello could not be coerced into a LocalTime")
+          )
+        },
+        test("-> LocalDateTime") {
+          val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
+          val coercion  = StandardType.StringType.coerce(StandardType.LocalDateTimeType(formatter))
+
+          assertTrue(
+            coercion.get("2020-12-31 12:34:56") == Right(
+              java.time.LocalDateTime.parse("2020-12-31 12:34:56", formatter)
+            ),
+            coercion.get("hello") == Left("String hello could not be coerced into a LocalDateTime")
+          )
+        },
+        test("-> ZonedDateTime") {
+          val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss Z")
+          val coercion  = StandardType.StringType.coerce(StandardType.ZonedDateTimeType(formatter))
+
+          assertTrue(
+            coercion.get("2020-12-31 12:34:56 +0100") == Right(
+              java.time.ZonedDateTime.parse("2020-12-31 12:34:56 +0100", formatter)
+            ),
+            coercion.get("hello") == Left("String hello could not be coerced into a ZonedDateTime")
+          )
+        },
+        test("-> Duration") {
+          val temporalUnit = ChronoUnit.SECONDS
+          val coercion     = StandardType.StringType.coerce(StandardType.Duration(temporalUnit))
+
+          assertTrue(
+            coercion.get("PT1S") == Right(java.time.Duration.of(1, temporalUnit)),
+            coercion.get("hello") == Left("String hello could not be coerced into a Duration")
+          )
+        },
+        test("-> Period") {
+          val coercion = StandardType.StringType.coerce(StandardType.PeriodType)
+
+          assertTrue(
+            coercion.get("P1Y2M3D") == Right(java.time.Period.of(1, 2, 3)),
+            coercion.get("hello") == Left("String hello could not be coerced into a Period")
```

Comment:
I suggest only testing that it `isLeft`, because that makes the tests less brittle and the error messages easier to improve.